src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,240 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            while (resultSet.next()) {
+                Long carsId = resultSet.getObject("id", Long.class);
+                Car car = parseCar(resultSet);
+                car.setId(carsId);
+                cars.add(car);
+            }
+
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't get all cars", throwables);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversFromCurrentCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?,"
+                   + "manufacturer_id = ? WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't update current car - " + car,
+                    throwables);
+        }
+        deleteDriversRelationsWithCar(car);
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() != 0;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't delete car by id " + id,
+                    throwables);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT id AS cars_id, model AS cars_model, manufacturer_id FROM cars c"
+                + " JOIN cars_drivers cd ON c.id = cd.cars_id"
+                + " WHERE cd.drivers_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            List<Car> cars = new ArrayList<>();
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(get(resultSet.getObject("cars_id", Long.class)).get());
```

Comment:
you stil open new connection here. First in this method, second in method get. For list without drivers here and after catch init drivers for your cars

Reply:
Okay, got in now